lib/stm32/common/usart_common_all.c

```diff
@@ -37,9 +37,9 @@ Devices can have up to 3 USARTs and 2 UARTs.
 /*---------------------------------------------------------------------------*/
 /** @brief USART Set Baudrate.
 
-The baud rate is computed from the APB high-speed prescaler clock (for
-USART1/6) or the APB low-speed prescaler clock (for other USARTs). These values
-must be correctly set before calling this function (refer to the
+The baud rate is computed from the APB slock associated with the USART/UART
+device. The RCC must be configured for desired frequency before this function
+is called or this may use the incorrect clock for calculations (refer to the
```

Comment:
I know you want uart working, but can you stop adding things with uarts when the commit titles are "rcc: add feature xxx" ? :)  add the functionality, then add a single standalone commit for the uart stuff itself.

Reply:
I'm sorry...I wanted to have a sample usage to make sure it links and what not...I will pull this out :-)

Reply:
I understand how it happens :)